doc/source/building.rst

```diff
@@ -65,7 +67,45 @@ If you wish to select the single mode, pass the ``--single`` option to
 configure, though note that this is the default. The reentrant mode is selected
 by passing the option ``--reentrant`` to configure.
 
-ABI and architecture support
+
+
+Architecture support
+-------------------------------------------------------------------------------
+
+In some cases, it is necessary to override the CPU/OS defaults. This can be
+done by passing ``--build=cpu-vendor-os`` to configure, for which 
+
+The choices for OS include ``Linux``, ``MINGW32``, ``MINGW64``, ``CYGWIN32``,
+``CYGWIN64``, ``Darwin``, ``FreeBSD``, ``OpenBSD``, ``SunOS`` and numerous other
+operating systems.
```

Comment:
Why not list them all explicitly? Conversely: does it make sense to list any for which there isn't special code? E.g. `SunOS` could be removed as it does nothing special, or does it?

I think for the user it's more important to get a complete list of actually supported OSes whose name makes a difference, and then perhaps allow a `Generic` standin to be used whenever the OS is not on the list?

Reply:
I believe it said something similar earlier, so I believe I just kept it.